pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/RRDD.scala

```diff
@@ -161,9 +169,31 @@ private abstract class BaseRRDD[T: ClassTag, U: ClassTag](
     }.start()
   }
 
-  protected def openDataStream(input: InputStream): Closeable
+  protected def readData(length: Int): U
+
+  protected def read(): U = {
+    try {
+      val length = dataStream.readInt()
 
-  protected def read(): U
+      length match {
+        case SpecialLengths.TIMING_DATA =>
+          // Timing data from R worker
+          logger.boot = dataStream.readDouble - logger.boot
+          logger.init = dataStream.readDouble
+          logger.broadcast = dataStream.readDouble
+          logger.input = dataStream.readDouble
+          logger.compute = dataStream.readDouble
+          logger.output = dataStream.readDouble
+          logInfo(logger.reportTime)
```

Comment:
We did not have any benefits from `TimeLogger`, I'd like to pull them here, it will be more readable.
```
val boot = dataStream.readDouble - bootTime
...
logInfo(xxx, boot, ...)
```